### PR TITLE
feat(testing): run tests against scp pg hyperledger

### PR DESCRIPTION
**Pull request by vobu**
this PR enables to run relevant tests against both the dockerized pg and the hyperscaler pg on cf.

for that, the included sample proj is packed up and deployed as `cap-proj-srv` (and expects a hyperscaler pg service instance named `cap-proj-database`). more details are described in `docs/CONTRIBUTING.md#run-tests-agains-scp-cf-postgresql-hyperledger-service`).

the essential switch of running either against dockerized pg or against both dockerized pg + hyperscaler pg is the presence of a `/__tests__/__assets__/cap-proj/.env`. default is to run against dockerized pg only.

### CODE REVIEW

**Comment by vobu** [comment]
also also affects #64 in the sense of providing a test env for
- dockerized pg
- hyperscaler pg

@d-sooter you might want to add the additional test env (different schema) into `__tests__/lib/pg/_buildSuiteEnvironments.js`

File: __tests__/__assets__/cap-proj/rest-client-test/requests.http [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# @fqdn = http://localhost:4004
+@fqdn = https://4d3dc3fctrial-dev-cap-proj-srv.cfapps.eu10.hana.ondemand.com
+###
+# "reset" aka redeploy from csv
+POST {{fqdn}}/beershop/reset
```
Comment (gregorwolf) [inline __tests__/__assets__/cap-proj/rest-client-test/requests.http:5]: With {{$dotenv scpServiceURL}} we could use a .env file located in the rest-client-test folder and not expose any trial subaccount that should not be exposed.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -163,6 +166,65 @@ $> npm run test:as-sqlite
 
 As stated above, the SQLite-based app is accessible via http://localhost:4004
 
+### run tests agains SCP cf PostgreSQL, hyperledger service
+
+SAP offers a [managed PostgreSQL instance on CF](https://discovery-center.cloud.sap/serviceCatalog/postgresql-hyperscaler-option) now. It is also possible to deploy the sample project included here in cds-pg (`__tests__/__assets__/cap-proj`) to SAP CP and connect it to the cf-based hyperscaler PostgreSQL service.
+
+_Note:_ all of the below will work in local \*nix-like environments only (Linux, macOS, WSL on Windows), not on Windows `cmd` or `powershell`.
+
+First, create an instance `cap-proj-database` of the hyperscaler PostgreSQL service.
```
Comment (gregorwolf) [inline docs/CONTRIBUTING.md:175]: I would suggest that we either provide the cf command or include the PostgreSQL service instance creation in the mta.yaml

File: __tests__/__assets__/cap-proj/scp-deployment/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (gregorwolf) [inline __tests__/__assets__/cap-proj/scp-deployment/deploy.sh:1]: To create all the files needed for SAP CP Cloud Foundry deployment we could use [cds-dbm build](https://github.com/mikezaschka/cds-dbm#build)

File: __tests__/__assets__/cap-proj/rest-client-test/requests.http (hunk 1 above)
Comment (vobu) [reply __tests__/__assets__/cap-proj/rest-client-test/requests.http:5]: 👍 will do

File: __tests__/__assets__/cap-proj/scp-deployment/deploy.sh (hunk 3 above)
Comment (vobu) [reply __tests__/__assets__/cap-proj/scp-deployment/deploy.sh:1]: removed the entire folder as it is not used for deployment
rather, cds deploy (via cds-pg's simple deploy mechanism) is used (from the -srv module) as we don't need delta deployments here

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (vobu) [reply docs/CONTRIBUTING.md:175]: 👍 will do

**Review by gregorwolf — APPROVED** [review]

**Pull request merged**